### PR TITLE
Add SwiftPM plugins for API Gateway client generation.

### PR DESCRIPTION
**Pull request by tachyonics**
*Issue #, if available:*

*Description of changes:*
Add SwiftPM plugins to generate a standalone client for an API Gateway based endpoint. 

As an example https://github.com/amzn/smoke-framework-examples/tree/api_gateway_client_generation/PersistenceExampleClient is generated by the command-

```
swift run APIGatewayClientGenerate --base-file-path ${basePath}/smoke-framework-examples/PersistenceExampleClient
```

Moves to using `swift-argument-parser` and a configuration file `api-gateway-client-swift-codegen.json` at the base path specified.

Uses the following default options-
1. modelFormat: .openAPI30
2. eventLoopFutureClientAPIs: .disabled
3. minimumCompilerSupport: .v5_6
4. clientConfigurationType:  .configurationObject

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift [hunk 1]
```diff
@@ -20,19 +20,35 @@ import ServiceModelCodeGeneration
...
+    case codeGenClient
+}
 
 public struct APIGatewayClientCodeGeneration {
     
     public static func generateFromModel<ModelType: ServiceModel>(
         modelFilePath: String,
         modelType: ModelType.Type,
+        generationType: GenerationType,
+        modelLocation: ModelLocation?,
```
Comment (pbthif) [inline Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift:39]: What's the difference between `modelFilePath` and `modelLocation`?

File: Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift [hunk 2]
```diff
@@ -42,43 +58,128 @@ public struct APIGatewayClientCodeGeneration {
...
-            baseName: applicationDescription.baseName,
-            isThrowingMock: true,
-            asyncAwaitAPIs: asyncAwaitAPIs)
-        let awsClientDelegate = APIGatewayClientDelegate(
-            baseName: applicationDescription.baseName,
-            asyncAwaitAPIs: asyncAwaitAPIs,
-            contentType: "application/json", signAllHeaders: false)
-        let awsModelErrorsDelegate = APIGatewayClientModelErrorsDelegate()
+                                                    generationType: GenerationType,
+                                                    modelLocation: ModelLocation?,
```
Comment (pbthif) [inline Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift:87]: This doesn't seem to be used in the function.

**Review by pbthif — APPROVED** [review]

File: Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift (hunk 1 above)
Comment (tachyonics) [reply Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift]: modelLocation is unused and has been removed.

File: Sources/APIGatewayClientGenerate/APIGatewayClientGenerateCommand.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    
+    private func getModelLocation(config: APIGatewayClientSwiftCodeGen) -> ModelLocation? {
+        // find the model for the current target
+        let filteredModelLocations = config.modelLocations?.targetMap.compactMap { (targetName, modelLocation) -> ModelLocation? in
+            if targetName == target {
+                return modelLocation
+            }
+            
+            return nil
+        }
```
[the remark this answers is not in the document]
Comment (tachyonics) [reply Sources/APIGatewayClientGenerate/APIGatewayClientGenerateCommand.swift]: fixed.

File: Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift (hunk 2 above)
Comment (tachyonics) [reply Sources/APIGatewayClientModelGenerate/APIGatewayClientCodeGeneration.swift]: yes, removed.

**Review by pbthif — APPROVED** [review]

**Pull request merged**